File: LEVEL2/가장 큰 수.java
import java.io.*;
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String[] str = new String[numbers.length];
        
        for(int i = 0; i < numbers.length; i++) {
            str[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(str, (str1, str2) -> (str2 + str1).compareTo(str1 + str2));
        
        if(str[0].equals("0")) {
            return "0";
        }
        
        StringBuilder answer = new StringBuilder();
        
        for(int i = 0; i < str.length; i++) {
            answer.append(str[i]);
        }
        
        return answer.toString();
    }
}

File: LEVEL2/가장 큰 수.cpp
#include<string>
#include<vector>
#include<algorithm>
using namespace std;

bool cmp(string a, string b) {
    return a + b > b + a;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> vc;
    
    for(auto i : numbers) {
        vc.push_back(to_string(i));
    }
    
    sort(vc.begin(), vc.end(), cmp);
    
    if(vc.at(0) == "0") {
        return "0";
    }
    
    for(auto i : vc) {
        answer += i;
    }
    
    return answer;
}